src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -53,24 +63,15 @@
     private final AtomicInteger connectionRecoveryAttempts = new AtomicInteger(0);
     private final AtomicBoolean connectionIsClosed = new AtomicBoolean(false);
     private final RabbitMQConfiguration configuration;
+    private final ResendMessageConfiguration resendConfiguration;
     private final String subscriberName;
     private final AtomicInteger nextSubscriberId = new AtomicInteger(1);
-
-    private final RecoveryListener recoveryListener = new RecoveryListener() {
-        @Override
-        public void handleRecovery(Recoverable recoverable) {
-            LOGGER.debug("Count tries to recovery connection reset to 0");
-            connectionRecoveryAttempts.set(0);
-            CommonMetrics.setRabbitMQReadiness(true);
-            LOGGER.debug("Set RabbitMQ readiness to true");
-        }
-
-        @Override
-        public void handleRecoveryStarted(Recoverable recoverable) {}
-    };
+    private final ThreadLocal<Map<Long, PublishData>> rejectHandlers = ThreadLocal.withInitial(ConcurrentHashMap::new);
+    private final ExecutorService resendRejectService;
```

Comment:
!! This service isn't shut down in the close method.

Reply:
Maybe create one ExecutorService for CommonFactory instance with minimal thread and share them for other entities.